### ISSUE TITLE
Property 'toJSON' does not exist on type 'BSModalContextBuilder'

### ISSUE THREAD

**Issue by ssashok10** [shlomiassaf/angular2-modal#158]
I am submitting a bug report,

I have implemented standard modal as well as custom modal dialog windows and it works fine. I follow the below sample code to implement custom dialog by using Angular 2 RC5 and angular2-modal 1.1.1.

Everything was working fine and i suddenly started getting the below error message and my visual studio is not even compiling the project because of this issue.

"angular2-modal" has no exported member 'ModalModule'. I have re-installed this module many times and re-started my machine as well. I keep getting this error. My code was a working before and i follow your sample below,

http://plnkr.co/edit/hwbmF9?p=preview

Please let me know how to fix this issue. Why my "app.module.ts" is not able to find 'ModalModule' suddenly? Thanks in advance.

**Comment by ssashok10** [comment]
I changed to the latest version 2.0.0-beta.3 and followed the approach mentioned on the below sample and everything works fine. 

http://plnkr.co/edit/mbPzd8?p=preview

As i am not seeing the same issue with the latest version, i am closing this issue. Thanks.

**Comment by ssashok10** [comment]
Its very strange that the working code started giving the below two errors,

1. Property 'toJSON' does not exist on type 'BSModalContextBuilder<CustomModalContext>'

2. Property 'open'' does not exist on type 'Modal'

Here is the code snippet,

```
 let overlayConfig: OverlayConfig = {
            context: builder.toJSON()
        };

        const dialog = this.modal.open(CustomModal, overlayConfig);
```

I am using the latest version 2.0.0-beta.3 with Angular 2 RC 5.  Can you please let me know what could be the reason i am seeing these build errors?

**Comment by shlomiassaf** [comment]
@ssashok10 Can you confirm this is solved with `2.0.0-beta.4`?

**Comment by ssashok10** [comment]
I have tried with 2.0.0-beta.9 and it looks good. Thanks for your help.
